demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/ZipResponseGenerator.php

```diff
@@ -0,0 +1,119 @@
+ *
+ * (c) 2010-present DEMOS plan GmbH, for more information see the license file.
+ *
+ * All rights reserved
+ */
+
+namespace demosplan\DemosPlanCoreBundle\Logic\FileResponseGenerator;
+
+use demosplan\DemosPlanCoreBundle\Entity\File;
+use demosplan\DemosPlanCoreBundle\Exception\InvalidDataException;
+use demosplan\DemosPlanCoreBundle\Logic\Procedure\NameGenerator;
+use demosplan\DemosPlanCoreBundle\Logic\ZipExportService;
+use demosplan\DemosPlanCoreBundle\Utilities\DemosPlanPath;
+use Exception;
+use PhpOffice\PhpSpreadsheet\Writer\Exception as WriterException;
+use PhpOffice\PhpSpreadsheet\Writer\IWriter;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use ZipStream\Exception\FileNotFoundException;
+use ZipStream\Exception\FileNotReadableException;
+use ZipStream\ZipStream;
+
+class ZipResponseGenerator extends FileResponseGeneratorAbstract
+{
+    public function __construct(
+        array $supportedTypes,
+        NameGenerator $nameGenerator,
+        private readonly ZipExportService $zipExportService,
+        private readonly LoggerInterface $logger,
+        private readonly TranslatorInterface $translator
+    ) {
+        parent::__construct($nameGenerator);
+        $this->supportedTypes = $supportedTypes;
+    }
+
+    public function __invoke(array $file): Response
+    {
+        return $this->createStreamedResponseForZip($file);
+    }
+
+    private function createStreamedResponseForZip(array $file): StreamedResponse
+    {
+        return $this->zipExportService->buildZipStreamResponse(
+            $file['zipFileName'].'.zip',
+            fn (ZipStream $zipStream) => $this->fillZipWithData($zipStream, $file)
+        );
+    }
+
+    private function fillZipWithData(ZipStream $zipStream, array $file): void
+    {
+        $errorMessages = [];
+        $temporaryFullFilePath = DemosPlanPath::getTemporaryPath($file['xlsx']['filename']);
+        /** @var IWriter $writer */
+        $writer = $file['xlsx']['writer'];
+        try {
+            $writer->save($temporaryFullFilePath);
+            $this->zipExportService->addFileToZipStream(
+                $temporaryFullFilePath,
+                $file['zipFileName'].'/'.$file['xlsx']['filename'],
+                $zipStream
+            );
+        } catch (FileNotFoundException|FileNotReadableException $e) {
+            $this->logger->critical('Unable to load or read file from path.', [$e]);
+            $errorMessages[] = $this->translator->trans('error.statements.zip.export.generic.xlsx');
+        } catch (WriterException|Exception $e) {
+            $this->logger->critical('An error occured during creation of zip file for export.', [$e]);
+            $errorMessages[] = $this->translator->trans('error.statements.zip.export.generic.xlsx');
+        }
+
+        foreach ($file['attachments'] as $attachmentArray) {
```

Comment:
key might be undefined

Reply:
considered in [f976dd0](https://github.com/demos-europe/demosplan-core/pull/2337/commits/f976dd016149e9ab8e9ac0347abfc98fb0143702)